core/src/main/java/io/questdb/griffin/engine/table/LatestByAllIndexedFilteredRecordCursor.java

```diff
@@ -79,51 +51,14 @@ public void close() {
     @Override
     protected void buildTreeMap(SqlExecutionContext executionContext) {
         filter.init(this, executionContext);
+        super.buildTreeMap(executionContext);
+        postProcessRows();
+    }
 
-        final int keyCount = getSymbolTable(columnIndex).size() + 1;
-
-        long keyLo = 0;
-        long keyHi = keyCount;
-
-        rows.extend(keyCount);
-        rows.setPos(rows.getCapacity());
-        rows.zero(0);
-
-        long rowCount = 0;
-        long argsAddress = LatestByArguments.allocateMemory();
-        LatestByArguments.setRowsAddress(argsAddress, rows.getAddress());
-        LatestByArguments.setRowsCapacity(argsAddress, rows.getCapacity());
-
-        DataFrame frame;
-        int frameColumnIndex = columnIndexes.getQuick(columnIndex);
-        while ((frame = this.dataFrameCursor.next()) != null && rowCount < keyCount) {
-            final BitmapIndexReader indexReader = frame.getBitmapIndexReader(frameColumnIndex, BitmapIndexReader.DIR_BACKWARD);
-            final long rowLo = frame.getRowLo();
-            final long rowHi = frame.getRowHi() - 1;
-            final int partitionIndex = frame.getPartitionIndex();
-
-            LatestByArguments.setKeyLo(argsAddress, keyLo);
-            LatestByArguments.setKeyHi(argsAddress, keyHi);
-            LatestByArguments.setRowsSize(argsAddress, rows.size());
-
-            BitmapIndexUtilsNative.latestScanBackward(
-                    indexReader.getKeyBaseAddress(),
-                    indexReader.getKeyMemorySize(),
-                    indexReader.getValueBaseAddress(),
-                    indexReader.getValueMemorySize(),
-                    argsAddress,
-                    indexReader.getUnIndexedNullCount(),
-                    rowHi, rowLo,
-                    partitionIndex, indexReader.getValueBlockCapacity()
-            );
-            rowCount += LatestByArguments.getRowsSize(argsAddress);
-            keyLo = LatestByArguments.getKeyLo(argsAddress);
-            keyHi = LatestByArguments.getKeyHi(argsAddress) + 1;
-        }
-        LatestByArguments.releaseMemory(argsAddress);
-
+    @Override
+    protected void postProcessRows() {
         rows.setPos(rows.getCapacity());
-        for(long r = 0; r < rows.getCapacity(); ++r) {
+        for (long r = 0; r < rows.getCapacity(); ++r) {
```

Comment:
in java more optimal loop would be:

```java
for (long r = 0, n = rows.getCapacity(); r < n; ++r)
```

this would end up calling method once vs possibly many times